gensim/models/ldaseqmodel.py

```diff
@@ -0,0 +1,1121 @@
+    log_norm_counts = log_norm_counts / sum(log_norm_counts)
+    log_norm_counts = numpy.log(log_norm_counts)
+    
+
+    # setting variational observations to transformed counts
+    for t in range(0, T):
+        sslm.obs[:,t] = log_norm_counts
+
+    # set variational parameters
+    sslm.obs_variance = obs_variance
+    sslm.chain_variance = chain_variance
+
+    # compute post variance
+    for w in range(0, W):
+       compute_post_variance(w, sslm, sslm.chain_variance)
+
+    for w in range(0, W):
+       compute_post_mean(w, sslm, sslm.chain_variance)
+
+    update_zeta(sslm)
+    compute_expected_log_prob(sslm)
+
+def init_ldaseq_ss(ldaseq, topic_chain_variance, topic_obs_variance, alpha, init_suffstats):
+
+    ldaseq.alphas = alpha
+    for k in range(0, ldaseq.num_topics):
+
+        sstats = init_suffstats[:,k]
+        sslm_counts_init(ldaseq.topic_chains[k], topic_obs_variance, topic_chain_variance, sstats)
+
+        # initialize the below matrices only if running DIM
+        # ldaseq.topic_chains[k].w_phi_l = numpy.zeros((ldaseq.num_terms, ldaseq.num_sequences))
+        # ldaseq.topic_chains[k].w_phi_sum = numpy.zeros((ldaseq.num_terms, ldaseq.num_sequences))
+        # ldaseq.topic_chains[k].w_phi_sq = numpy.zeros((ldaseq.num_terms, ldaseq.num_sequences))
+
+def fit_lda_seq(ldaseq, seq_corpus):
+
+    K = ldaseq.num_topics
+    W = ldaseq.num_terms
+    data_len = seq_corpus.num_sequences
+    num_docs = seq_corpus.num_docs
+    
+    LDA_INFERENCE_MAX_ITER = 25
+
+    bound = 0
+    heldout_bound = 0
+    ldasqe_em_threshold = 1e-4
+    convergence = ldasqe_em_threshold + 1
+
+    iter_ = 0
+    final_iters_flag = 0 
+    last_iter = 0
+
+    # this is a flag/input do something about it
+    lda_seq_min_iter = 6
+    lda_seq_max_iter = 20
+    
+    while iter_ < lda_seq_min_iter or ((final_iters_flag is 0 or convergence > ldasqe_em_threshold) and iter_ <= lda_seq_max_iter):
+        if not (iter_ < lda_seq_min_iter or ((final_iters_flag is 0 or convergence > ldasqe_em_threshold) and iter_ <= lda_seq_max_iter)):
+            last_iter = 1
+
+        # log
+        print (" EM iter " , iter_)
+        print ("E Step")
+
+        # writing to file
+        em_log.write(str(bound) + "\t" + str(convergence))
+        old_bound = bound
+
+        # initiate sufficient statistics
+        topic_suffstats = []
+        for k in range(0, K):
+            topic_suffstats.append(numpy.resize(numpy.zeros(W * data_len), (W, data_len)))
+
+        # set up variables
+        gammas = numpy.resize(numpy.zeros(num_docs * K), (num_docs, K))
+        lhoods = numpy.resize(numpy.zeros(num_docs * K + 1), (num_docs, K + 1))
+
+        bound = lda_seq_infer(ldaseq, seq_corpus, topic_suffstats, gammas, lhoods, iter_, last_iter)
+
+        print ("M Step")
+
+        topic_bound = fit_lda_seq_topics(ldaseq, topic_suffstats)
+        bound += topic_bound
+
+
+        if ((bound - old_bound) < 0):
+            if (LDA_INFERENCE_MAX_ITER == 1):
```

Comment:
what is the purpose of this?
also it can be less lines if it is a `dict` or just multiply by 2